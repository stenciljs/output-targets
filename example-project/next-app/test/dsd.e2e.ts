import { browser, expect } from '@wdio/globals';
import prettier from 'prettier';

describe('Static Declarative Shadow DOM (DSD)', () => {
  for (const component of ['simple', 'nested', 'nested-with-react'/*, 'complex-use-client'*/]) {
    it(`should render DSD of '${component}' correctly`, async () => {
      await browser.url(`/dsd/${component}`);
      const page = await fetch(await browser.getUrl());
      const rawContent = await page.text();
      const componentHtml = rawContent.match(/<main[^>]*>([\s\S]*)<\/main>/)?.[1] ?? '';
      const formattedHtml = await prettier.format(componentHtml, { parser: 'html' });
      // Depeneding on the order the components are first rendered, the ids might be different
      const sanitizedHtml = formattedHtml
        .replaceAll(/r\.\d+/g, 'r.X')
        .replaceAll(/t\.\d+/g, 't.X')
        .replaceAll(/s-id="\d+/g, 's-id="X')
        .replaceAll(/c-id="\d+/g, 'c-id="X');

      expect(sanitizedHtml).toMatchSnapshot();
    });
  }
});
