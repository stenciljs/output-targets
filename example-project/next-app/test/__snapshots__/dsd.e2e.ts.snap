// Snapshot v1

exports[`Static Declarative Shadow DOM (DSD) > should render DSD of 'nested' correctly 1`] = `
"<my-list class="hydrated" s-id="X"
  ><template shadowrootmode="open">
    <ul c-id="X.0.0.0">
      <slot c-id="X.1.1.0"></slot></ul></template
  ><my-list-item class="hydrated" s-id="X"
    ><template shadowrootmode="open">
      <li c-id="X.0.0.0">
        <slot c-id="X.1.1.0"></slot></li></template
    >Item 1</my-list-item
  ><my-list-item class="hydrated" s-id="X"
    ><template shadowrootmode="open">
      <li c-id="X.0.0.0">
        <slot c-id="X.1.1.0"></slot></li></template
    >Item 2</my-list-item
  ><my-list-item class="hydrated" s-id="X"
    ><template shadowrootmode="open">
      <li c-id="X.0.0.0">
        <slot c-id="X.1.1.0"></slot></li></template
    >Item 3</my-list-item
  ></my-list
>
"
`;

exports[`Static Declarative Shadow DOM (DSD) > should render DSD of 'nested-with-react' correctly 1`] = `
"<my-list class="hydrated" s-id="X"
  ><template shadowrootmode="open">
    <ul c-id="X.0.0.0">
      <slot c-id="X.1.1.0"></slot></ul></template
  ><my-list-item class="hydrated" s-id="X"
    ><template shadowrootmode="open">
      <li c-id="X.0.0.0">
        <slot c-id="X.1.1.0"></slot></li
    ></template>
    <div
      style="
        padding: 1rem;
        border: 1px solid black;
        border-radius: 0.5rem;
        margin: 1rem;
      "
    >
      <p style="margin-bottom: 0.5rem">Pure React Component</p>
      <button
        style="border: 1px solid black; padding: 0.5rem; border-radius: 0.5rem"
      >
        Click me (Clicked:
        <!-- -->0<!-- -->)
      </button>
    </div></my-list-item
  ></my-list
>
"
`;

exports[`Static Declarative Shadow DOM (DSD) > should render DSD of 'simple' correctly 1`] = `
"<my-button
  class="button button-solid hydrated my-activatable my-focusable"
  href="#"
  s-id="X"
  ><template shadowrootmode="open">
    <a c-id="X.0.0.0" class="button-native" href="#">
      <span c-id="X.1.1.0" class="button-inner">
        <slot c-id="X.2.2.0" name="icon-only"></slot>
        <slot c-id="X.3.2.1" name="start"></slot>
        <slot c-id="X.4.2.2"></slot>
        <slot c-id="X.5.2.3" name="end"></slot>
      </span> </a></template
  >Test</my-button
>
"
`;
